test/assertions_test.rb

```diff
@@ -87,13 +87,43 @@ def test_assert_statsd_call
       end
     end
 
+    assert_no_assertion_triggered do
+      foo_1_metric_tags = { times: 2, tags: ['foo:1'] }
+      foo_2_metric_tags = { times: 1, tags: ['foo:2'] }
+      @test_case.assert_statsd_increments('counter', [foo_1_metric_tags, foo_2_metric_tags]) do
+        StatsD.increment('counter', tags: { foo: 1 })
+        StatsD.increment('counter', tags: { foo: 1 })
+        StatsD.increment('counter', tags: { foo: 2 })
+      end
+    end
```

Comment:
I think thats a question for whoever is writing the test .. they may want to test that metrics pop up in a certain order .. or they might not care. I haven't seen any tests so far that care about ordering so far. Implementing it with ordering was easy. With respect to number of occurrences, I figured I'd just extend the approach that was already in place .. allow devs to specify the `times` a metric should be seen .. rather than forcing them to repeat asserted metric option sets.